### PR TITLE
add support for GRAPH.CONSTRAINT

### PR DESCRIPTION
**Pull request by OfirMos**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `$ tox` pass with this change (including linting)?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [x] Is the new or changed code fully tested?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Is there an example added to the examples folder (if applicable)?
- [x] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

_Please provide a description of the change here._

### CODE REVIEW

File: redis/commands/graph/commands.py [hunk 1]
```diff
@@ -214,6 +217,26 @@ def explain(self, query, params=None):
         plan = self.execute_command(EXPLAIN_CMD, self.name, query)
         return ExecutionPlan(plan)
 
+    def constraint(
+        self,
+        key: KeyT,
+        op: str,
+        type: str,
```
Comment (chayim) [inline redis/commands/graph/commands.py:224]: I'd rather not use type as an input, as type is a semi-reserved word in python

File: redis/commands/graph/commands.py [hunk 2]
```diff
@@ -214,6 +217,26 @@ def explain(self, query, params=None):
         plan = self.execute_command(EXPLAIN_CMD, self.name, query)
         return ExecutionPlan(plan)
 
+    def constraint(
+        self,
+        key: KeyT,
+        op: str,
```
Comment (chayim) [inline redis/commands/graph/commands.py:223]: IMHO OP should be more descriptive (operation)

File: redis/commands/graph/commands.py [hunk 3]
```diff
@@ -214,6 +217,26 @@ def explain(self, query, params=None):
...
+        type: str,
+        entity_type: str,
+        label_name: str,
+        n_props: int,
```
Comment (chayim) [inline redis/commands/graph/commands.py:227]: ```num_props```

File: redis/commands/graph/commands.py [hunk 4]
```diff
@@ -214,6 +217,26 @@ def explain(self, query, params=None):
...
+        props: List[str]):
```
Comment (chayim) [inline redis/commands/graph/commands.py:228]: ```properties```

File: redis/commands/graph/commands.py (hunk 3 above)
Comment (chayim) [inline redis/commands/graph/commands.py:227]: ```num_props```

File: redis/commands/graph/commands.py (hunk 4 above)
Comment (chayim) [inline redis/commands/graph/commands.py:228]: ```properties```

File: redis/commands/graph/commands.py [hunk 5]
```diff
@@ -214,6 +217,26 @@ def explain(self, query, params=None):
...
+        """
+        Constraint operation,
+        For more information see `GRAPH.CONSTRAINT <https://redis.io/commands/graph.constraint>`_. # noqa
```
Comment (chayim) [inline redis/commands/graph/commands.py:231]: can I see the graph.constraint docs that are proposed? it's a dead link

File: redis/commands/graph/commands.py [hunk 6]
```diff
@@ -214,6 +217,26 @@ def explain(self, query, params=None):
...
+        self,
+        key: KeyT,
+        op: str,
+        type: str,
+        entity_type: str,
+        label_name: str,
+        n_props: int,
+        props: List[str]):
+        """
+        Constraint operation,
```
Comment (chayim) [inline redis/commands/graph/commands.py:230]: Describe what a constraint operation is. These docs force people to go to the module docs - but we need to help the client users themselves.

File: redis/commands/graph/commands.py (hunk 5 above)
Comment (OfirMos) [reply redis/commands/graph/commands.py:231]: It won't be when we merge it

File: redis/commands/graph/commands.py (hunk 6 above)
Comment (OfirMos) [reply redis/commands/graph/commands.py:230]: Once we will have the docs ready I'll copy the description here.

File: redis/commands/graph/commands.py (hunk 5 above)
Comment (chayim) [reply redis/commands/graph/commands.py]: @OfirMos did this happne, or can we close off the PR? I don't see it merged or available currently.

**Pull request closed without merging**